packages/xrpl/src/utils/getNFTokenID.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable @typescript-eslint/consistent-type-assertions -- Necessary for parsing metadata */
+import { flatMap } from 'lodash'
+
+import {
+  CreatedNode,
+  isCreatedNode,
+  isModifiedNode,
+  ModifiedNode,
+  TransactionMetadata,
+} from '../models/transactions/metadata'
+
+interface NFToken {
+  NFToken: {
+    NFTokenID: string
+    URI: string
+  }
+}
+
+/**
+ * Gets the NFTokenID for an NFT recently minted with NFTokenMint.
+ *
+ * @param meta - Metadata from the response to submitting an NFTokenMint transaction.
+ * @returns The NFTokenID for the minted NFT.
+ * @throws if meta is not TransactionMetadata.
+ */
+export default function getNFTokenID(
+  meta: TransactionMetadata,
+): string | undefined {
+  /* eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- Provides a nicer error for js users */
+  if (meta.AffectedNodes === undefined) {
+    throw new TypeError(`Unable to parse the parameter given to getNFTokenID. 
+    'meta' must be the metadata from an NFTokenMint transaction. Received ${JSON.stringify(
```

Comment:
I think you can pass the data in the second parameter instead of stringifying it

Reply:
Seems like that's not the case for TypeError - I get a TypeScript error when trying to just pass in a second argument to augment the message -
```
src/utils/getNFTokenID.ts:33:7 - error TS2554: Expected 0-1 arguments, but got 2.

33       meta,
         ~~~~
```

Reply:
Have the serialized last to make it easier to read.  Maybe just `Recieved: ` and remove `instead.`